src/Symfony/Component/Validator/Constraints/AllValidator.php

```diff
@@ -11,39 +11,14 @@
 
 namespace Symfony\Component\Validator\Constraints;
 
-use Symfony\Component\Validator\Constraint;
-use Symfony\Component\Validator\ConstraintValidator;
-use Symfony\Component\Validator\Exception\UnexpectedTypeException;
-use Symfony\Component\Validator\Exception\UnexpectedValueException;
-
 /**
  * @author Bernhard Schussek <bschussek@gmail.com>
+ *
+ * @api
+ *
+ * @deprecated Deprecated in 4.3, to be removed in 5.0. Use
+ *             {@link \Symfony\Component\Validator\Constraints\EachValidator} instead.
  */
-class AllValidator extends ConstraintValidator
+class AllValidator extends EachValidator
```

Comment:
Should we swap the inheritance chain so that the new classes can be passed anywhere where the old classes are expected without breaking anything?